torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -162,6 +162,10 @@ ProcessGroupAgent::ProcessGroupAgent(
   }
 }
 
+ProcessGroupAgent::~ProcessGroupAgent() {
+  localShutdown();
```

Comment:
This can throw an exception which would be dangerous. Add a try-catch and just LOG(ERROR) the exception here.